### PR TITLE
Use shared informers in scheduler and token controller

### PR DESCRIPTION
**Pull request by smarterclayton**
Contains upstream fixes and a few minor reaction changes. Rebase pick was a little ugly because of the number of changes, but verified it passed in upstream branch and downstream.

@deads2k @sttts [test]

### CODE REVIEW

File: pkg/cmd/server/start/start_master.go [hunk 1]
```diff
@@ -609,6 +609,9 @@ func startControllers(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) erro
 	// Start these first, because they provide credentials for other controllers' clients
 	oc.RunServiceAccountsController()
 	oc.RunServiceAccountTokensController(controllerManagerOptions)
+	// Must start a few informers now
+	oc.Informers.KubernetesInformers().Start(utilwait.NeverStop)
```
Comment (deads2k) [inline pkg/cmd/server/start/start_master.go]: These need to start to make the SA token controller active, right?  Be more descriptive in your comment.  Lots of people end up here.

Comment (smarterclayton) [inline pkg/cmd/server/start/start_master.go]: Yeah, will add more of a comment

**Comment by deads2k** [comment]
lgtm

**Comment by smarterclayton** [comment]
[merge] for great performance

**Comment by smarterclayton** [comment]
[merge][severity: blocker] memory use is still p0 for 3.6

**Pull request merged**